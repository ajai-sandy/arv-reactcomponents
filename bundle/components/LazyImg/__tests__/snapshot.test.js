import React from "react";
import { mount, shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import TestComponent from "../test.component";

describe("Root Component Tests", () => {
  let mounted;
  beforeEach(() => {
    mounted = mount(<TestComponent />);
  });

  it("Matches Shallow SnapShot", () => {
    expect(shallowToJson(shallow(<TestComponent />))).toMatchSnapshot();
  });

  it("Matches Mounted SnapShot", done => {
    setTimeout(() => {
      mounted.update();
      global.window.dispatchEvent(new Event("scroll"));

      setTimeout(() => {
        mounted.update();
        expect(mounted).toMatchSnapshot();

        done();
      }, 1000);
    }, 300);
  });

  it("Unmount", () => {
    mounted.unmount();
  });
});
