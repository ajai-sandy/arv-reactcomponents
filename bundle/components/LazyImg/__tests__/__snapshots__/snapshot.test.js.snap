// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Component Tests Matches Mounted SnapShot 1`] = `
<LazyImgExample>
  <div>
    <LazyImg
      alt="NNNOW"
      className="nw-carouselexample-img1"
      index={0}
      offset={300}
      onError={[Function]}
      onLoad={[Function]}
      onWinLoad={true}
      parentElement={null}
      src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
    >
      <div
        className="nwc-lazyimg-container nw-carouselexample-img1"
      >
        <img
          alt="NNNOW"
          className="nwc-lazyimg "
          onError={[Function]}
          onLoad={[Function]}
          src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
        />
      </div>
    </LazyImg>
    <LazyImg
      alt="NNNOW"
      className="nw-carouselexample-img2"
      index={0}
      offset={300}
      onError={[Function]}
      onLoad={[Function]}
      onWinLoad={false}
      parentElement={null}
      src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
    >
      <div
        className="nwc-lazyimg-container nw-carouselexample-img2"
      >
        <img
          alt="NNNOW"
          className="nwc-lazyimg "
          onError={[Function]}
          onLoad={[Function]}
          src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
        />
      </div>
    </LazyImg>
    <LazyImg
      alt="NNNOW"
      className="nw-carouselexample-img3"
      index={0}
      offset={0}
      onError={[Function]}
      onLoad={[Function]}
      onWinLoad={false}
      parentElement={
        <div>
          <div
            class="nwc-lazyimg-container nw-carouselexample-img1"
          >
            <img
              class="nwc-lazyimg "
              src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
              alt="NNNOW"
            />
          </div>
          <div
            class="nwc-lazyimg-container nw-carouselexample-img2"
          >
            <img
              class="nwc-lazyimg "
              src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
              alt="NNNOW"
            />
          </div>
          <div
            class="nwc-lazyimg-container nw-carouselexample-img3"
          />
          <div
            class="nwc-lazyimg-container nw-carouselexample-img4"
          >
            <img
              class="nwc-lazyimg "
              src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
              alt="NNNOW"
            />
          </div>
        </div>
      }
      src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
    >
      <div
        className="nwc-lazyimg-container nw-carouselexample-img3"
      />
    </LazyImg>
    <LazyImg
      alt="NNNOW"
      className="nw-carouselexample-img4"
      index={0}
      offset={300}
      onError={[Function]}
      onLoad={[Function]}
      onWinLoad={false}
      parentElement={
        <div>
          <div
            class="nwc-lazyimg-container nw-carouselexample-img1"
          >
            <img
              class="nwc-lazyimg "
              src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
              alt="NNNOW"
            />
          </div>
          <div
            class="nwc-lazyimg-container nw-carouselexample-img2"
          >
            <img
              class="nwc-lazyimg "
              src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
              alt="NNNOW"
            />
          </div>
          <div
            class="nwc-lazyimg-container nw-carouselexample-img3"
          />
          <div
            class="nwc-lazyimg-container nw-carouselexample-img4"
          >
            <img
              class="nwc-lazyimg "
              src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
              alt="NNNOW"
            />
          </div>
        </div>
      }
      src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
    >
      <div
        className="nwc-lazyimg-container nw-carouselexample-img4"
      >
        <img
          alt="NNNOW"
          className="nwc-lazyimg "
          onError={[Function]}
          onLoad={[Function]}
          src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
        />
      </div>
    </LazyImg>
  </div>
</LazyImgExample>
`;

exports[`Root Component Tests Matches Shallow SnapShot 1`] = `
<div>
  <LazyImg
    alt="NNNOW"
    className="nw-carouselexample-img1"
    index={0}
    offset={300}
    onError={[Function]}
    onLoad={[Function]}
    onWinLoad={true}
    parentElement={null}
    src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
  />
  <LazyImg
    alt="NNNOW"
    className="nw-carouselexample-img2"
    index={0}
    offset={300}
    onError={[Function]}
    onLoad={[Function]}
    onWinLoad={false}
    parentElement={null}
    src="https://static.nnnow.com/client/assets/images/nnnow-logo-beta_1.png"
  />
</div>
`;
